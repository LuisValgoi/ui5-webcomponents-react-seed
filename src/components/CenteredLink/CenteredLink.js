import React from 'react';

import { FlexBox } from '@ui5/webcomponents-react/lib/FlexBox';
import { FlexBoxAlignItems } from '@ui5/webcomponents-react/lib/FlexBoxAlignItems';
import { FlexBoxDirection } from '@ui5/webcomponents-react/lib/FlexBoxDirection';
import { FlexBoxJustifyContent } from '@ui5/webcomponents-react/lib/FlexBoxJustifyContent';
import { Link } from '@ui5/webcomponents-react/lib/Link';
import { LinkDesign } from '@ui5/webcomponents-react/lib/LinkDesign';

import './CenteredLink.css';

function CenteredLink({ text, onClick }) {
  return (
    <FlexBox
      style={{ width: '100%', height: 'auto' }}
      direction={FlexBoxDirection.Column}
      justifyContent={FlexBoxJustifyContent.Center}
      alignItems={FlexBoxAlignItems.Center}
    >
      <Link
        onClick={onClick}
        design={LinkDesign.Emphasized}>
        {text}
      </Link>
    </FlexBox>
  );
}

export default CenteredLink;
