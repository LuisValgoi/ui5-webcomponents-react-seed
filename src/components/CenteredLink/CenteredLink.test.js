import React from 'react';

import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import CenteredLink from './CenteredLink';

test('should match snapshot', () => {
  const { asFragment } = render(<CenteredLink />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders with test text', () => {
  const { getByText } = render(<CenteredLink text='test' />);
  const linkElement = getByText('test');
  expect(linkElement).toBeInTheDocument();
});
