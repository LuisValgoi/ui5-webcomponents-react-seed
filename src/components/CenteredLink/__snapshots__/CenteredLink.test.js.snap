// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="FlexBox-flexBox-0-2-1 FlexBox-flexBoxDirectionColumn-0-2-8 FlexBox-justifyContentCenter-0-2-5 FlexBox-alignItemsCenter-0-2-14 FlexBox-flexWrapNoWrap-0-2-17"
    style="width: 100%; height: auto;"
  >
    <ui5-link
      design="Emphasized"
    />
  </div>
</DocumentFragment>
`;
